ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -239,15 +239,15 @@ let resync_host ~__context ~host =
               (Ref.string_of self) ;
             Xapi_clustering.Daemon.enable ~__context ;
             maybe_switch_cluster_stack_version ~__context ~self ~cluster_stack ;
+            let verify = Stunnel_client.get_verify_by_default () in
+            set_tls_config ~__context ~self ~verify ;
             (* Note that join_internal and enable both use the clustering lock *)
             Client.Client.Cluster_host.enable ~rpc ~session_id ~self
           ) ;
           (* create the watcher here so that the watcher exists after toolstack restart *)
           create_cluster_watcher_on_master ~__context ~host ;
           Xapi_observer.initialise_observer ~__context
-            Xapi_observer_components.Xapi_clusterd ;
-          let verify = Stunnel_client.get_verify_by_default () in
-          set_tls_config ~__context ~self ~verify
+            Xapi_observer_components.Xapi_clusterd
```

Comment:
May I know more background of the fix, as I see the plan to fix this issue is:
```
1. move set_tls_config somewhere else, maybe closer to when we start up the cluster daemon

2. when restarting the server (i.e. set_tls_config), we should not allow other remote operations to happen, hence a lock would be useful here
```
Is the 2nd plan considered in this fix?

Reply:
no, the second fix is much more complicated. This PR is only for the first fix